kernel/include/syscall_handler.h

```diff
@@ -125,6 +125,116 @@ extern void _thread_perms_all_clear(struct k_thread *thread);
  */
 void _k_object_uninit(void *obj);
 
+/**
+ * @brief Obtain the size of a C string passed from user mode
+ *
+ * Given a C string pointer and a maximum size, obtain the true
+ * size of the string (not including the trailing NULL byte) just as
+ * if calling strnlen() on it, with the same semantics of strnlen() with
+ * respect to the return value and the maxlen parameter.
+ *
+ * Any memory protection faults triggered by the examination of the string
+ * will be safely handled and an error code returned.
+ *
+ * NOTE: Doesn't guarantee that user mode has actual access to this
+ * string, you will need to still do a Z_SYSCALL_MEMORY_READ()
+ * with the obtained size value to guarantee this.
+ *
+ * @param src String to measure size of
```

Comment:
We should mention that this function must be called with interrupts disabled, due to interaction with fault handler.

Reply:
I don't believe this is actually necessary.
The fixup pointer is stored on a per-thread basis.
We can context switch out unless I'm missing something.

Reply:
What about situation when z_user_string_nlen() is being interrupted by a higher priority interrupt which triggers a fault? By my understanding, we will get a false positive here - instead of crashing kernel, we will assume that fault was caused by user string checking.

Reply:
Ah I see now.
Will address in next iteration of this PR, I have an idea on how to do this without requiring interrupt locking.

Reply:
New approach taken, interrupt locking should no longer be necessary.

Reply:
Now it looks better.